src/main/java/adris/altoclef/tasks/resources/wood/CollectSticksTask.java

```diff
@@ -0,0 +1,139 @@
+import java.util.Optional;
+
+public class CollectSticksTask extends ResourceTask {
+
+    private final int _targetCount;
+
+    private final Item[] _itemsToPickup;
+
+    public CollectSticksTask(int targetCount) {
+        super(Items.STICK, targetCount);
+        _targetCount = targetCount;
+        List<Item> itemsToPickup = new ArrayList<>(Arrays.asList(ItemHelper.LOG));
+        itemsToPickup.addAll(Arrays.asList(ItemHelper.PLANKS));
+        itemsToPickup.add(Items.DEAD_BUSH);
+        _itemsToPickup = itemsToPickup.toArray(new Item[itemsToPickup.size()]);
+    }
+
+    @Override
+    protected boolean shouldAvoidPickingUp(AltoClef mod) {
+        return false;
+    }
+
+    @Override
+    protected void onResourceStart(AltoClef mod) {
+        mod.getBehaviour().push();
+        mod.getBlockTracker().trackBlock(Blocks.DEAD_BUSH);
+        mod.getBlockTracker().trackBlock(ItemHelper.itemsToBlocks(ItemHelper.LOG));
+        mod.getBlockTracker().trackBlock(ItemHelper.itemsToBlocks(ItemHelper.PLANKS));
+    }
+
+    @Override
+    protected Task onResourceTick(AltoClef mod) {
+        // Calculate how much sticks can be made from planks and craft
+        int potentialStickCountFromPlanks = (mod.getItemStorage().getItemCount(ItemHelper.PLANKS)/2)*4 + mod.getItemStorage().getItemCount(Items.STICK);
+        if (potentialStickCountFromPlanks >= _targetCount) {
+            for (Item plank : ItemHelper.PLANKS) {
+                int count = mod.getItemStorage().getItemCount(plank);
+                if (count > 0) {
+                    setDebugState("Crafting sticks from planks");
+                    return new CraftInInventoryTask(new RecipeTarget(Items.STICK, (count/2)*4, generateStickRecipe(ItemHelper.PLANKS)));
+                }
+            }
+        }
+        // Do the same for logs to craft planks
+        int potentialStickCountFromLogsAndPlanks = potentialStickCountFromPlanks + mod.getItemStorage().getItemCount(ItemHelper.LOG)*8;
+        if (potentialStickCountFromLogsAndPlanks >= _targetCount) {
+            for (Item log : ItemHelper.LOG) {
+                int count = mod.getItemStorage().getItemCount(log);
+                if (count > 0) {
+                    Item plankCheck = ItemHelper.logToPlanks(log);
+                    if (plankCheck == null) {
+                        Debug.logError("Invalid/Un-convertable log: " + log + " (failed to find corresponding plank)");
+                    }
+                    int plankCount = mod.getItemStorage().getItemCount(plankCheck);
+                    int otherPlankCount = mod.getItemStorage().getItemCount(ItemHelper.PLANKS) - plankCount;
+                    int targetTotalPlanks = Math.min(count*4 + plankCount, _targetCount - otherPlankCount);
+                    setDebugState("Crafting planks from logs for sticks");
+                    return new CraftInInventoryTask(new RecipeTarget(plankCheck, targetTotalPlanks, generatePlankRecipe(ItemHelper.LOG)));
```

Comment:
This approach works, you might be able to simplify it though with something like this:
```java
if (we have enough planks to craft our remaining sticks) {
    craft sticks like you're doing now
}
if (we have logs) {
    return TaskCatalogue.getItemTask("planks");
}
```

Reply:
This is also unencesarry because logs and planks are handled in altoclef elsewhere. As a result, this code is either never going to run OR will be replacing code that already works. We only need to decide if it is time to use planks for crafting instead of hunting bush.